### ISSUE TITLE
Remote validator cache behaviour 

### ISSUE THREAD

**Issue by ankurk91** [guillaumepotier/Parsley.js#1181]
What kind of issue is this? (put 'x' between the square brackets)

 - [x] Feature Request. Make sure there's no good way to do what you want first;
       consider asking on http://stackoverflow.com/questions/ask first.

**Take this example** -
I want coupon code to be validated in real-time, all coupons has an expiry date and max usage limit.
```html
<div class="form-group">
                  <label class="control-label">Coupon code</label>
                  <input type="text" class="form-control" id="coupon-code" name="coupon[code]"
                         placeholder="Enter Coupon Code"
                         minlength="3"
                         data-parsley-success-class="has-success"
                         data-parsley-trigger="input"
                         data-parsley-remote
                         data-parsley-remote-validator="coupon"
                         data-parsley-remote-message="This coupon is invalid."
                         data-parsley-remote-options='{"type": "POST", "dataType": "json", "cache":"false"}'
                         data-parsley-debounce="500">
</div>
```
```js
 Parsley.addAsyncValidator('coupon', function (jqXHR) {
    console.log(jqXHR.responseText);
    var message = 'This coupon is invalid.';
    var discount = 0;
    switch (jqXHR.status) {
      case 0:
        message = 'No internet connection. Verify Network.';
        break;
      case 200:
        // Update coupon amount coming from server
        discount = JSON.parse(jqXHR.responseText).data.discount;
        break;
      case 422:
        message = window.app.util.parseValidationErrors(jqXHR.responseText);
    }
    // Update coupon discount on view
   $('.coupon-discount').text(parseFloat(discount));
    // Set validation message coming from server
    this.$element.parsley({remoteMessage: message});
    // Field is valid only when status code is 200
    return 200 === jqXHR.status;

  }, '/validate-coupon.php');
```
Suppose user enter ```FREE50``` and found it is a valid coupon code but later on (before submit) he tries some more coupons, lastly he goes with ```FREE50```, in the meantime this coupon expires but Parsley will still show that it is a valid coupon.
Parsley cache remote responses and does not hit server again if the user input is already validated.
I thought that ```cache:false``` will work here, but not
```
 data-parsley-remote-options='{"type": "POST", "dataType": "json", "cache":"false"}'
```
jQuery [$.ajax](http://api.jquery.com/jquery.ajax/) has an option disable cache. But Parsley does not respect it.
All I want Parsley to validate user inputs in real-time on user input.

PR #1074 caught my attention last night. I didn't go with a PR, I want to discuss it first here. May be i am the only one who has such scenario.

**Comment by cristiangiagante** [comment]
Same issue, is not enough with clean cache on submits, we need disable cache all time if we need it. Please fix this.

**Comment by marcandre** [comment]
I understand the issue.

The cache can be very helpful while the user types in a field.

I'm wondering if there should be a setting to disregard the cache, or if it shouldn't always be disregarded when submitting.

**Comment by marcandre** [comment]
Not sure it's the very best solution, but let's clear the cache before form submission.

Released 2.7.2.
